mongo_service.go

```diff
@@ -101,3 +90,30 @@ func (service *MongoService) unlockDb() {
 	}
 	info.Println("DB UNLOCK command successfully executed.")
 }
+
+func isNodeForBackup(mongoJson map[string]interface{}) (bool, mongoClusterHostInfo) {
+	if mongoJson["me"] == nil || mongoJson["hosts"] == nil {
+		warn.Println("Node is not part of a cluster.")
+		return false, mongoClusterHostInfo{}
+	}
+
+	master := mongoJson["ismaster"].(bool)
+
+	thisNode := mongoJson["me"].(string)
+	primary := mongoJson["primary"].(string)
+
+	allNodesUntyped := mongoJson["hosts"].([]interface{})
+	clusterSize := len(allNodesUntyped)
+	allNodes := make([]string, 0, clusterSize)
+	for _, v := range allNodesUntyped {
+		node := v.(string)
+		if node != primary {
+			allNodes = append(allNodes, node)
+		}
+	}
+	sort.Strings(allNodes)
+	lowestSecondary := allNodes[0]
```

Comment:
I know that is unlikely to happen, but if there are no secondary nodes this will fail I think...